cumulusci/util/xml/test/test_salesforce_encoding.py

```diff
@@ -0,0 +1,68 @@
+from glob import glob
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from io import StringIO
+
+from lxml import etree
+import pytest
+
+from cumulusci.util.xml.salesforce_encoding import serialize_xml_for_salesforce
+
+
+class TestSalesforceEncoding:
+    def test_xml_declaration(self):
+        xml = etree.parse(StringIO("<foo/>"))
+        print("xml", xml)
+        out = serialize_xml_for_salesforce(xml, xml_declaration=True)
+        assert out.startswith("<?xml")
+
+        out = serialize_xml_for_salesforce(xml, xml_declaration=False)
+        assert not out.startswith("<?xml")
+
+        out = serialize_xml_for_salesforce(xml)
+        assert out.startswith("<?xml")
+
+    def test_roundtripping(self):
+        from cumulusci.tasks.metadata import tests
+
+        print("Q", Path(tests.__file__).parent / "sample_package.xml")
+
+        print("r", Path(tests.__file__).parent / "sample_package.xml")
+
+        files = glob(str(Path(tests.__file__).parent / "sample_package.xml"))
+
+        #   If you fiddle with the salesforce encoder, the code below may be
+        #   useful to ensure that it faithfully round-trips, but it only works
+        #   if run in a directory with a parent-directory which contains a few
+        #   CCI projects which contain translations in them. At the time of
+        #   writing it roundtripped 421 files without a byte of difference.
+        #
+        # files = glob("../*/*/*ranslations/**ranslation", recursive=True)
+
+        assert files
+
+        for file in files:
+            orig = open(file).read()
+            tree = etree.parse(file)
+            out = serialize_xml_for_salesforce(tree)
+            print(f"[{orig}]")
+            print(f"[{out}]")
+            assert orig == out
```

Comment:
Please add an assertion error with the name of the file (or this might be an opportunity to use pytest's test parametrization feature, but that could be more trouble than it's worth)

Reply:
3d4b7068